### PR TITLE
add drf39 and drfmaster tests

### LINKED ISSUE #486: prepare for DRF 3.9 release
[DRF release 3.9](https://github.com/encode/django-rest-framework/milestone/66) is [coming soon](https://www.encode.io/reports/september-2018/). We should get ready to deal with any upstream changes.

### PR DESCRIPTION
**Pull request by n2ygk**
Fixes #486 

## Description of the Change

Add test matrix entries for drf3.9 and drf master branch (so we can get an early look at 3.10+).

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [x] documentation updated
- [x] changelog entry added to `CHANGELOG.md`
- [x] author name in `AUTHORS`

### CODE REVIEW

**Comment by n2ygk** [comment]
@sliverc Yes, I'll add py37 and django21

**Comment by n2ygk** [comment]
Will need to figure out what the magic incantation is to get Python 3.7 made available in Travis:
```
3.7 is not installed; attempting download
Downloading archive: https://s3.amazonaws.com/travis-python-archives/binaries/ubuntu/14.04/x86_64/python-3.7.tar.bz2
$ curl -sSf -o python-3.7.tar.bz2 ${archive_url}
curl: (22) The requested URL returned error: 403 Forbidden
Unable to download 3.7 archive. The archive may not exist. Please consider a different version.
```
FWIW all the tox Python 3.7 stuff passed locally using MacOS Mojave, Xcode10 and `pyenv install -v 3.7.0` (although not super-smoothly - https://github.com/pyenv/pyenv/issues/1219)

**Comment by n2ygk** [comment]
@sliverc I'm at a loss here... It looks like even though the distro is xenial, travis is still looking in the trusty archive (14.04 instead of 16.04).

**Comment by n2ygk** [comment]
I'll try adding "sudo: required" and see if that fixes it. See https://blog.travis-ci.com/2018-10-04-combining-linux-infrastructures

**Comment by n2ygk** [comment]
@sliverc Ready for (hopefully final) review (and merge).

File: .travis.yml [hunk 1]
```diff
@@ -35,6 +50,18 @@ matrix:
       env: TOXENV=py35-df20-django20-drf37
     - python: 3.5
       env: TOXENV=py35-df20-django20-drf38
+    - python: 3.5
+      env: TOXENV=py35-df20-django20-drf39
+    - python: 3.5
+      env: TOXENV=py35-df20-django20-drfmaster
+    - python: 3.5
+      env: TOXENV=py35-df20-django21-drf37
```
Comment (sliverc) [inline .travis.yml]: Django 2.1 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 2]
```diff
@@ -35,6 +50,18 @@ matrix:
...
+    - python: 3.5
+      env: TOXENV=py35-df20-django21-drf38
```
Comment (sliverc) [inline .travis.yml]: Django 2.1 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 3]
```diff
@@ -46,6 +73,51 @@ matrix:
       env: TOXENV=py36-df20-django20-drf37
     - python: 3.6
       env: TOXENV=py36-df20-django20-drf38
+    - python: 3.6
+      env: TOXENV=py36-df20-django20-drf39
+    - python: 3.6
+      env: TOXENV=py36-df20-django20-drfmaster
+    - python: 3.6
+      env: TOXENV=py36-df20-django21-drf37
```
Comment (sliverc) [inline .travis.yml]: Django 2.1 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 4]
```diff
@@ -46,6 +73,51 @@ matrix:
...
+    - python: 3.6
+      env: TOXENV=py36-df20-django21-drf38
```
Comment (sliverc) [inline .travis.yml]: Django 2.1 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 5]
```diff
@@ -46,6 +73,51 @@ matrix:
...
+    - python: 3.6
+      env: TOXENV=py36-df20-django21-drf39
+    - python: 3.6
+      env: TOXENV=py36-df20-django21-drfmaster
+
+    - python: 3.7
+      dist: xenial
+      sudo: required
+      env: TOXENV=py37-df20-django20-drf37
```
Comment (sliverc) [inline .travis.yml]: Python 3.7 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 6]
```diff
@@ -46,6 +73,51 @@ matrix:
...
+    - python: 3.7
+      dist: xenial
+      sudo: required
+      env: TOXENV=py37-df20-django20-drf38
```
Comment (sliverc) [inline .travis.yml]: Python 3.7 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 7]
```diff
@@ -46,6 +73,51 @@ matrix:
...
+      sudo: required
+      env: TOXENV=py37-df20-django20-drf39
+    - python: 3.7
+      dist: xenial
+      sudo: required
+      env: TOXENV=py37-df20-django20-drfmaster
+    - python: 3.7
+      dist: xenial
+      sudo: required
+      env: TOXENV=py37-df20-django21-drf37
```
Comment (sliverc) [inline .travis.yml]: Python 3.7 is only supported with DRF 3.9 so we should remove this.

File: .travis.yml [hunk 8]
```diff
@@ -46,6 +73,51 @@ matrix:
...
+    - python: 3.7
+      dist: xenial
+      sudo: required
+      env: TOXENV=py37-df20-django21-drf38
```
Comment (sliverc) [inline .travis.yml]: Python 3.7 is only supported with DRF 3.9 so we should remove this.

File: tox.ini [hunk 9]
```diff
@@ -1,16 +1,20 @@
 [tox]
 envlist =
-    py27-df11-django111-drf{36,37,38}
-    py{34,35,36}-df20-django111-drf{36,37,38},
-    py{34,35,36}-df20-django20-drf{37,38},
+    py27-df11-django111-drf{36,37,38,39}
+    py{34,35,36}-df20-django111-drf{36,37,38,39,master},
+    py{34,35,36,37}-df20-django20-drf{37,38,39,master},
+    py{35,36,37}-df20-django21-drf{37,38,39,master},
```
Comment (sliverc) [inline tox.ini]: This needs to be adjusted to cover the above requirements of matrix.

File: .travis.yml (hunk 1 above)
Comment (n2ygk) [reply .travis.yml]: You don't think it's a good idea to test master and get an early heads-up (e.g. on dot-releases of 3.9)?

**Comment by n2ygk** [comment]
@sliverc You don't think it's a good idea to test against master in order to get an early heads-up as 3.9 dot releases become available? It's an allowed failure so won't break travis builds. Personally I would have felt better about not having to rush to get drf39 tests added here after 3.9 was published. Luckily nothing broke.

**Comment by sliverc** [comment]
@n2ygk 
My comment was not concerning drfmaster tests (those need to be tested against combos which are officially supported by the previous release I guess) but the test with official DRF releases.

Example:
a combo Django 2.1 + Python 3.6 and DRF 3.7 should not be tested in our matrix as DRF states that Django 2.1 is only supported from DRF 3.9 onward.

This way we can reduce the size of the matrix and also tell the user what is officially supported when the look at the CI build and a specific DRF version.

Hope this clarifies.

**Comment by sliverc** [comment]
Thanks. LGTM. Merging.

**Pull request merged**